### DISCUSSION TITLE
[vcpkg source][Download ZIP] Command "vcpkg x-history [package]" cannot execute.

### DISCUSSION THREAD

**Discussion by artzers** [microsoft/vcpkg#20735] [Ideas]
**Describe the bug**
The command "vcpkg x-history [package]" cannot execute.

**Environment**
- OS: [Windows 10/ ubuntu 16]
- Compiler: VS 2017, gcc/g++ 8.4

**To Reproduce**
1. ./vcpkg x-history [any package]
2. See error

**Expected behavior**
Shows the full history of CONTROL version of a port, including the vcpkg commit hash, CONTROL version, and date of vcpkg commit. 

**Failure logs** 
Windows10: run "./vcpkg x-history vtk"
"Failed to run command: "D:\vcpkg-master\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --git-dir="D:\vcpkg-master/.git" log --format="%H %cd" --date=short --left-only -- ports/vtk/."

Ubuntu: run "./vcpkg x-history qt5"
"Failed to run command: "/usr/bin/git" --git-dir="/home/super/Downloads/vcpkg-master/.git" log --format="%H %cd" --date=short --left-only -- ports/qt5/."

**Additional context**
1. The git program runs OK.
2. I have tried to use administrator rights. The command cannot be executed.

**Comment by JackBoosY** [comment]
Hi @artzers, thanks for reporting this issue!
Did you use powershell? I can't repro this issue on Windows 10 or Ubuntu now.

Can you run `D:\vcpkg-master\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --git-dir="D:\vcpkg-master.git" log --format="%H %cd" --date=short --left-only -- ports/vtk/.` manually and provide me the results?

Thanks.

**Comment by artzers** [comment]
@JackBoosY  First I use CMD to run the command:  D:\vcpkg-master\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe --git-dir="D:\vcpkg-master\\.git" log --format="%H %cd" --date=short --left-only -- ports/qt5/
 The result is : fatal: not a git repository: 'D:\vcpkg-master\\.git'

Then I use powershell to run the command. The result is : fatal: not a git repository: 'D:\vcpkg-master\\.git'

**Comment by JackBoosY** [comment]
@artzers Can you try to re-clone a new vcpkg and retry again?

**Comment by artzers** [comment]
@JackBoosY  I have downloaded the vcpkg zip from github, decompressed and built again. The same error still occured. The default git source is PortableGit-2.26.2-32-bit.7z.exe.

D:\vcpkg>vcpkg x-history vtk
A suitable version of git was not found (required v2.26.2). Downloading portable git v2.26.2...
Extracting git...
Failed to run command: "D:\vcpkg\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --git-dir="D:\vcpkg\\.git" log --format="%H %cd" --date=short --left-only -- ports/vtk/.

D:\vcpkg>"D:\vcpkg\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --git-dir="D:\vcpkg\\.git" log --format="%H %cd" --date=short --left-only -- ports/vtk/
fatal: not a git repository: 'D:\vcpkg\\.git'

**Comment by JackBoosY** [comment]
@artzers You need to clone vcpkg using command `git clone https://github.com/microsoft/vcpkg.git` instead of download zip package.

**Comment by artzers** [comment]
@JackBoosY   Thanks 👍 . I git clone vcpkg, build and run x-history successfully.  
Commands "vcpkg.exe x-history vtk" and "vcpkg.exe x-history qt5-base" run correctly.
 version          date    vcpkg commit
               9.0-4    2020-07-02    2ad4074aea2415e962153544ca942942e7d52323
               9.0-3    2020-06-23    03766d0834c1ca8dea25ffafbe4fe1c60786a412
.........

However, I met error when running command "vcpkg.exe x-history qt5"
Failed to run command: "D:\vcpkg-test\vcpkg\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --git-dir="D:\vcpkg-test\vcpkg\.git" show 890a81bebc3642454e3833c789a35ba9a7989c9d:ports/qt5/CONTROL

I donot think it is a bug, but may confuse many beginners.

**Comment by PhoebeHui** [comment]
Double confirmed with latest souce, it works as expect when git clone vcpkg, however, it doesn't work when download vcpkg zip from github. I think it's a vcpkg feature.

```
C:\Users\phoebe\Downloads\vcpkg-master\vcpkg-master>vcpkg x-history zlib
A suitable version of git was not found (required v2.32.0). Downloading portable git v2.32.0...
Downloading git...
  https://github.com/git-for-windows/git/releases/download/v2.32.0.windows.2/PortableGit-2.32.0.2-32-bit.7z.exe -> C:\Users\phoebe\Downloads\vcpkg-master\vcpkg-master\downloads\PortableGit-2.32.0.2-32-bit.7z.exe
Extracting git...
A suitable version of 7zip was not found (required v19.0.0). Downloading portable 7zip v19.0.0...
Downloading 7zip...
  https://www.7-zip.org/a/7z1900-x64.msi -> C:\Users\phoebe\Downloads\vcpkg-master\vcpkg-master\downloads\7z1900-x64.msi
Extracting 7zip...
             version          date    vcpkg commit
```

**Reply by autoantwort** [reply]
vcpkg will never work when you only download it as zip. The versioning relies on the underlying git repo and that you can checkout specific trees.
